src/foam/nanos/mrac/ElectoralServiceServer.js

```diff
@@ -0,0 +1,231 @@
+    'java.util.concurrent.ThreadLocalRandom',
+    'java.util.Date',
+    'java.util.List',
+    'static foam.mlang.MLang.*'
+  ],
+
+  properties: [
+    {
+      name: 'state',
+      class: 'Enum',
+      of: 'foam.nanos.mrac.ElectoralServiceState',
+      value: 'foam.nanos.mrac.ElectoralServiceState.IN_SESSION'
+    },
+    {
+      name: 'electionTime',
+      class: 'DateTime'
+    },
+    {
+      name: 'votes',
+      class: 'Int',
+      value: 0
+    },
+    {
+      name: 'currentSeq',
+      class: 'Int',
+      value: 0
+    },
+    {
+      name: 'winner',
+      class: 'FObjectProperty',
+      of: 'foam.nanos.mrac.ClusterConfig'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'hasQuorum',
+      type: 'Boolean',
+      args: [
+        {
+          name: 'total',
+          type: 'Integer',
+        }
+      ],
+      javaCode: `
+      return getVotes() >= (total / 2 + 1);
+    `
+    },
+    {
+      name: 'recordResult',
+      synchronized: true,
+      args: [
+        {
+          name: 'result',
+          type: 'Integer',
+        },
+        {
+          name: 'config',
+          type: 'foam.nanos.mrac.ClusterConfig',
+        }
+      ],
+      javaCode: `
+      setVotes(getVotes() + 1);
+      if ( result > getCurrentSeq() ) {
+        setCurrentSeq(result);
+        setWinner(config);
+      }
+    `
+    },
+    {
+      name: 'buildURL',
+      args: [
+        {
+          name: 'config',
+          type: 'foam.nanos.mrac.ClusterConfig'
+        }
+      ],
+      type: 'String',
+      javaCode: `
+      try {
+        // TODO: protocol - http will do for now as we are behind the load balancers.
+        java.net.URI uri = new java.net.URI("http", null, config.getId(), config.getPort(), "/service/electoralService", null, null);
+        return uri.toURL().toString();
+      } catch (java.net.MalformedURLException | java.net.URISyntaxException e) {
+        ((Logger) getX().get("logger")).error(e);
+        return ""; // TODO: 
+      }
+      `
+    },
+    {
+      documentation: `Find the cluster configuration for 'this' (localhost) node.`,
+      name: 'findConfig',
+      type: 'foam.nanos.mrac.ClusterConfig',
+      args: [
+        {
+          name: 'x',
+          type: 'Context'
+        }
+      ],
+      javaCode: `
+      DAO dao = (DAO) x.get("clusterConfigDAO");
+      String hostname = System.getProperty("hostname");
+      if ( SafetyUtil.isEmpty(hostname) ) {
+        hostname = "localhost";
+      }
+      return ((ClusterConfig) dao.find_(x, hostname));
+      `
+    },
+    {
+      name: 'dissolve',
+      javaCode: `
+    ClusterConfig config = findConfig(getX());
+    if ( getState().equals(ElectoralServiceState.IN_SESSION) ) {
+      setElectionTime(new Date());
+      setState(ElectoralServiceState.ELECTION);
+      recordResult(vote(getElectionTime()), config);
+
+      List arr = (ArrayList) ((ArraySink) ((DAO) getX().get("clusterConfigDAO"))
+        .where(
+          AND(
```

Comment:
Also, only Secondaries and Primary can vote. Other replica nodes are not involved in voting.